src/core/core.controller.js

```diff
@@ -838,6 +838,35 @@ class Chart {
 		}
 	}
 
+	_updateVisibility(datasetIndex, index, visible) {
+		const me = this;
+		const mode = visible ? 'show' : 'hide';
+		const meta = me.getDatasetMeta(datasetIndex);
+		const anims = meta.controller._resolveAnimations(undefined, mode);
+		if (index === undefined) {
+			me.setDatasetVisibility(datasetIndex, visible);
+			// Animate visible state, so hide animation can be seen
+			anims.update(meta, {visible});
+		} else {
+			if (!meta.data[index]) {
+				return;
+			}
+			me.setDataVisibility(datasetIndex, index, visible);
+		}
+		meta._skipUpdate = true;
+		me.update();
```

Comment:
Why don't we just not call `update` instead of adding `_skipUpdate`?

Reply:
Added some comments to explain

Reply:
It feels a little off to me that we're reaching inside the meta object to set a private property `_skipUpdate`. E.g. it's not an API our users would be able to use really. What do you think about if we made `mode` also able to take a `function` to be resolved and then we could return a different `mode` for `datasetIndex`?

Reply:
done